lib/phoenix_component/declarative.ex

```diff
@@ -10,7 +10,10 @@ defmodule Phoenix.Component.Declarative do
     :inner_block,
     :myself,
     :flash,
-    :socket
+    :socket,
+    :uploads,
+    :streams,
+    :live_action
```

Comment:
@chrismccord i am worried about increasing this list because it makes each `:rest` operation more expensive. We need to handle `@__changed__`, `@__defaults__`, `@__slot__`, and `@inner_block`, because those are set automatically by the engine. But everything else will only show you up if you are doing `<.foo {assigns} />`, which is going to put spurious entries on `rest` anyway? So I would keep this list tailored only to the four above (with `@myself` and `@socket` if we want to be extra safe).

Reply:
Although this commit explicitly added flash: https://github.com/phoenixframework/phoenix_live_view/commit/c6f20b7f3d6a42ae9e4b8588152862a465b81a43 - do you remember why?

@bemesa21 what this means is that I think we need two lists. One is like `@reserved_assigns`, which we do not pass forward to components, and `@non_assignables` (or similar).